tensordict/nn/common.py

```diff
@@ -38,6 +38,21 @@ class FunctionalModuleWithBuffers:
 ]
 
 
+def _is_iterable_of_str(sequence_of_str: Sequence[str]) -> bool:
+    if not sequence_of_str:
+        return False
+    if isinstance(sequence_of_str, str):
+        return False
+    if not isinstance(sequence_of_str, (tuple, list, set)):
+        return False
+    if not all(
+        key and (isinstance(key, str) or _is_iterable_of_str(key))
```

Comment:
We want to allow `Sequence[Union[str, Tuple[str, ...]]` right? This recursive call allows arbitrary nesting, so for example

```python
>>> _is_iterable_of_str([(("too", "much", "nesting"),)])
True
```

Why not use the already existing function `_check_all_nested`?

Reply:
so just one level of nesting. right.
_check_all_nested is missing some checks. but I agree we can combine them